src/auxiliary/t8code.jl

```diff
@@ -131,47 +210,86 @@ function trixi_t8_count_interfaces(forest)
 
     return (interfaces = local_num_conform,
             mortars = local_num_mortars,
-            boundaries = local_num_boundary)
+            boundaries = local_num_boundary,
+            mpi_interfaces = local_num_mpi_conform,
+            mpi_mortars = local_num_mpi_mortars)
 end
 
-function trixi_t8_fill_mesh_info(forest, elements, interfaces, mortars, boundaries,
-                                 boundary_names)
-    # Check that forest is a committed, that is valid and usable, forest.
-    @assert t8_forest_is_committed(forest) != 0
+function trixi_t8_fill_mesh_info(mesh, elements, interfaces, mortars, boundaries,
+                                 boundary_names; mpi_mesh_info = undef)
+    @assert t8_forest_is_committed(mesh.forest) != 0
 
-    # Get the number of local elements of forest.
-    num_local_elements = t8_forest_get_local_num_elements(forest)
-    # Get the number of ghost elements of forest.
-    num_ghost_elements = t8_forest_get_num_ghosts(forest)
-    # Get the number of trees that have elements of this process.
-    num_local_trees = t8_forest_get_num_local_trees(forest)
+    num_local_elements = t8_forest_get_local_num_elements(mesh.forest)
+    num_local_trees = t8_forest_get_num_local_trees(mesh.forest)
 
     current_index = t8_locidx_t(0)
 
+    max_level = t8_forest_get_maxlevel(mesh.forest) #UInt64
+    max_tree_num_elements = UInt64(2^ndims(mesh))^max_level
+
+    if mpi_isparallel()
+        #! format: off
+        remotes = t8_forest_ghost_get_remotes(mesh.forest)
+        ghost_num_trees = t8_forest_ghost_num_trees(mesh.forest)
+
+        ghost_remote_first_elem = [num_local_elements +
+                                   t8_forest_ghost_remote_first_elem(mesh.forest, remote)
+                                   for remote in remotes]
+
+        ghost_tree_element_offsets = [num_local_elements +
+                                      t8_forest_ghost_get_tree_element_offset(mesh.forest, itree)
+                                      for itree in 0:(ghost_num_trees - 1)]
+
+        ghost_global_treeids = [t8_forest_ghost_get_global_treeid(mesh.forest, itree)
+                                for itree in 0:(ghost_num_trees - 1)]
+        #! format: on
+    end
```

Comment:
These things are used below. Do we need another `mpi_isparallel()` there?

Reply:
Only in MPI parallel mode. I'll refactor this routine anyway as was requested in another review comment.